pkg/service/test/test.go

```diff
@@ -274,7 +275,26 @@ func (t *tester) Test(path string, testReportPath string, generateTestReport boo
 		testSuiteNames = append(testSuiteNames, testSuiteName)
 	}
 
-	sort.Strings(testSuiteNames)
+	sort.SliceStable(testSuiteNames, func(i, j int) bool {
+
+		test1 := strings.Split(testSuiteNames[i], "-")
+		lenEl := len(test1)-1
+
+		num1, err := strconv.Atoi(test1[lenEl])
+		if err != nil {
+			return false
+		}
+
+		test2 := strings.Split(testSuiteNames[j], "-")
+		lenEl = len(test2)-1
+
+		num2, err := strconv.Atoi(test2[lenEl])
+		if err != nil {
+			return false
+		}
+
+		return num1 < num2
```

Comment:
a better way of doing thing could be 
```
parts1 := strings.Split(testSuiteNames[i], "-")
    parts2 := strings.Split(testSuiteNames[j], "-")

    if len(parts1) < 3 || len(parts2) < 3 {
        // Handle the case where there are not enough parts after the split
        return false // or true, or some other logic that makes sense for your use case
    }

    num1, err1 := strconv.Atoi(parts1[2])
    num2, err2 := strconv.Atoi(parts2[2])

    if err1 != nil || err2 != nil {
        // Handle the error case where the conversion from string to int fails
        return false // or true, or some other logic that makes sense for your use case
    }

    return num1 < num2
    
    ```
    
    Offcourse with names that are more suited for our context.

Reply:
case where length is less than 3 should be directly compare the string instead of returning false?